plugins/legion-hid2/fu-legion-hid2-device.c

```diff
@@ -117,6 +118,25 @@ fu_legion_hid2_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static void
+fu_legion_hid2_device_setup_child(FuDevice *device)
+{
+	g_autoptr(FuDevice) child = fu_legion_hid2_child_device_new(device);
+	g_autoptr(GError) error_child = NULL;
+
+	if (!fu_device_probe(child, &error_child)) {
+		g_info("%s", error_child->message);
+		return;
+	}
+
+	if (!fu_device_setup(child, &error_child)) {
+		g_info("%s", error_child->message);
+		return;
+	}
```

Comment:
does this need to be explicit? is the logic "don't add as a child if it's going to fail"?

Reply:
can we stick some of that in a comment for our future selves please?

Reply:
sure, I'll put it in a comment for the function